java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -323,12 +323,38 @@ public void testBitVector() {
       // Ensure unallocated space returns 0
       assertEquals(0, accessor.get(3));
 
+      // unset the previously set bits
       m.set(1, 0);
       m.set(1022, 0);
-
+      // this should set all the array to 0
       assertEquals(1024, accessor.getNullCount());
 
+      // set all the array to 1
       for (int i = 0; i < 1024; ++i) {
+        assertEquals(1024 - i, accessor.getNullCount());
+        m.set(i, 1);
+      }
+
+      assertEquals(0, accessor.getNullCount());
+
+      // re-allocate the vector with a new (odd) size, smaller by more than a byte than the original vector
+      // this allows to ensure that we use one byte less memory than the previous one
+      vector.allocateNew(1015);
+
+      // ensure it has been zeroed
+      assertEquals(1016, accessor.getNullCount());
```

Comment:
That was the question I asked on slack the other day. I didn't get a clear answer so I assumed that it should return the allocated size as defined in https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/BitVector.java#L113

There is an issue however if we want to get the actual length of the array. The length (`valueCount`) we pass through `allocateNew` is never set into the instance's `valueCount`, we set only the allocated size and `valueCount` is kept at zero. Is it the correct behaviour? If not, I should open another issue to fix that, change the implementation of `getNullCount` so that it uses `valueCount` rather than `allocationSizeInBytes` (the attribute that is set when we allocate the vector).